src/main/java/com/adamk33n3r/runelite/watchdog/ui/notifications/panels/NotificationPanel.java

```diff
@@ -109,6 +114,48 @@ public NotificationPanel(Notification notification, NotificationsPanel parentPan
         rightActions.setBackground(ColorScheme.DARKER_GRAY_COLOR);
         nameWrapper.add(rightActions, BorderLayout.EAST);
 
+        JPanel afkTimerConfigRow = new JPanel(new GridLayout(1, 2));
+        afkTimerConfigRow.setBorder(new EmptyBorder(4, 10, 0, 5));
+        afkTimerConfigRow.setBackground(ColorScheme.DARKER_GRAY_COLOR);
+        JLabel afkTimerLabel = new JLabel("AFK Seconds:");
+        afkTimerLabel.setToolTipText("Number of seconds for which the client doesn't get any mouse or keyboard inputs.");
+
+        JSpinner afkTimerSpinner = PanelUtils.createSpinner(Math.max(notification.getFireWhenAFKForSeconds(), 1),
+            1,
+            25 * 60,
+            1,
+            (val) -> {
+                notification.setFireWhenAFKForSeconds(val);
+                onChangeListener.run();
+            });
+        if (notification.getFireWhenAFKForSeconds() != 0) {
+            afkTimerConfigRow.add(afkTimerLabel);
+            afkTimerConfigRow.add(afkTimerSpinner);
+        }
+
+        JButton afkButton = PanelUtils.createToggleActionButton(
+            AFK_ICON,
+            AFK_ICON_HOVER,
+            NON_AFK_ICON,
+            NON_AFK_ICON_HOVER,
+            "Switch to only fire notification when you have been AFK for a certain amount of time",
+            "Enable notification even when you are active",
+            notification.getFireWhenAFKForSeconds() != 0,
+            (btn, modifiers) -> {
+                notification.setFireWhenAFKForSeconds(btn.isSelected() ? 5 : 0);
```

Comment:
Would probably be nice to add a default to the config like most notifications have and then set that here instead of just 5.

Would also be nice if toggling it on and off would remember your previous value.

Reply:
Okay I'll add the additional config for default value.

I can also make it remember the previous value, but then I'll need to add a new property to a notification. I liked how a single integer could both be the enabled state and the value but now they'll need to be separate. I'm not sure if that's worth it. It's not like someone will be toggling this on and off repeatedly. When editing the JSON by hand, it'd definitely be nicer to have one property that controls this setting than two.

Let me know what you want me to do here. I'll leave it as-is for now.

Reply:
I was just thinking making the property in the notification panel only. Not perpetual remembrance. But just if you're like toggling it back and forth on that screen. I think if you leave the screen, it's ok for it to reset.

Reply:
Got it! I'll do that. I'm strangely having a lot of trouble with injecting the `WatchdogConfig` into the `NotificationPanel`. Do you happen to know if there's some Java thing that makes it so you cannot inject into abstract classes?

Reply:
Um, ya it's not specifically a java thing, but injections into any sort of base class doesn't work actually. I forgot I had run into that before. The only solution I had found was you had to manually inject it in every child class. But that's nasty, so now I just use `WatchdogPlugin.getInstance().getConfig()`

Reply:
Awesome that works great. Thanks!